wagtail/exceptions.py

```diff
@@ -7,3 +7,12 @@ class PageClassNotFoundError(ImportError):
     """
 
     pass
+
+
+class BlockNormalizationError(TypeError):
```

Comment:
I'd question whether we really need a dedicated exception class for this - it occurs when people use an incompatible type when passing a value to a block, and to me that sounds like a job for a plain TypeError. Having a subclass is effectively just saying "a TypeError occurred during the normalization step", and that's leaking an unnecessary implementation detail to the user - for most purposes developers don't need to know that there's a normalization step.

Reply:
Ok, I'd put it in thinking it _might_ be useful to distinguish between "this is an unexpected type error" and "normalize was called with the wrong arguments" at recursive call sites, but in practice those are the same thing or it's not really needed. Removed in 6b6550ba06